src/backend/catalog/heap.c

```diff
@@ -1318,7 +1318,8 @@ heap_create_with_catalog(const char *relname,
 	 * it doesn't hurt to hold AccessExclusiveLock.  Do it here so callers
 	 * can't accidentally vary in their lock mode or acquisition timing.
 	 */
-	LockRelationOid(relid, AccessExclusiveLock);
+	if (!is_enr)
+		LockRelationOid(relid, AccessExclusiveLock);
```

Comment:
This is the case that was missed in the original PR

Reply:
Thanks! quick question, did the original PR have all tests passing before we merged? How come we didnt hit the issue before merge?

Reply:
Yeah, the original PR had all tests passing. I'm not sure why the crash didn't appear at that time.

Reply:
approving since Jason confirmed the crash was hit during table_variable_xact_nested test, which confirmed passes with this fix, and crashes without it